lib/common/mem.h

```diff
@@ -46,6 +46,8 @@ MEM_STATIC void MEM_check(void) { MEM_STATIC_ASSERT((sizeof(size_t)==4) || (size
 /*-**************************************************************
 *  Basic Types
 *****************************************************************/
+#ifndef BASIC_TYPES_DEFINED
+#define BASIC_TYPES_DEFINED
```

Comment:
This macro now acts as a kind of "hidden interface contract" with `mem.h`, which spells trouble for future maintenance.
That's one reason I prefer it removed.